pkg/datastoresrp/processors/mongodatabases/processor.go

```diff
@@ -32,7 +32,7 @@ type Processor struct {
 // Process implements the processors.Processor interface for Mongo database resources. It  validates Mongo database properties
 // and applies the values from the RecipeOutput.
 func (p *Processor) Process(ctx context.Context, resource *datamodel.MongoDatabase, options processors.Options) error {
-	validator := processors.NewValidator(&resource.ComputedValues, &resource.SecretValues, &resource.Properties.Status.OutputResources)
+	validator := processors.NewValidator(&resource.ComputedValues, &resource.SecretValues, &resource.Properties.Status.OutputResources, resource.ResourceMetadata().Status.Recipe)
```

Comment:
Could you help me understand why does it need to be `resource.ResourceMetadata().Status.Recipe` instead of `&resource.Properties.Status.Recipe`?